rfc/mediachain-rfc-4.md

```diff
@@ -0,0 +1,450 @@
+# RFC 4: New Directions in Mediachain: Protocols and Architecture
+
+Status: WIP WIP WIP WIP
+
+Author: vyzo
+
+## Lessons from Phase I: It's not a Blockchain
+
+The blockchain is an attractive solution: the public ledger promotes
+open data and discovery, while also simplifying streaming for online
+processing.
+
+But it comes with significant costs:
+
+* Complexities of maintaining consensus in a peer to peer network.
+* PoS/PoW economics that don't quite work in our domain.
+* Inefficient querying, requires the entire blockchain for read operations.
+* Difficult to scale to large datasets with bulk and firehose ingestion, the blockchain becomes the bottleneck.
+* Ultimately trying to solve a problem we don't have: double spending.
+
+In reality, we don't need a singular linearly ordered view of the world.
+And we want to allow the system to grow to very large datasets (billions
+of objects) supporting bulk and firehose ingestion models.
+
+Fundamentally, what we need is a map of media identifiers to sets
+of statements that allows us to track and relate media objects.
+This has the hallmarks of a CRDT data structure, which allows a system to
+achieve eventually consistent state without the need for consensus. 
+
+So what we want to build is a distributed database that supports
+upserts, connecting statements to one or more domain specific identifiers.
```

Comment:
> jesse: "define upsert"

Upsert: an idempotent insert/update operation that does not require pre-querying for existing records, and does not rely on knowing the state of the system